src/serde/timestamp.rs

```diff
@@ -0,0 +1,135 @@
+use crate::{IonError, Timestamp};
+use chrono::{DateTime, FixedOffset, Utc};
+use serde::de::Error;
+use serde::{self, de, ser, Deserialize, Serialize};
+use serde_with::{DeserializeAs, SerializeAs};
+use std::cell::Cell;
+use std::fmt;
+
+thread_local! {
+    /// Cell that contains a flag to determine when serialization and deserialization
+    /// is occurring with an Ion serializer. This allows us to know when to encode
+    /// Timestamps as ion timestamps
+    pub(crate) static ION_BINARY: Cell<bool> = Cell::new(false);
```

Comment:
Can you provide more explanation or a link to some documentation about why this is necessary? Maybe this is a naive question, but why is this a concern in the first place? (And why does it only apply to binary?)

Also, it looks like this flag is also used in the `decimal` module, so perhaps it should belong in `mod.rs` rather than in `timestamp.rs`.